### PR TITLE
Fix various VS code analysis complaints (and bugs)

### PR DESCRIPTION
**Pull request by Jalle19**
ping @FernetMenta @afedchin @ksooo since this changes your code and quite possibly the behavior.

### CODE REVIEW

**Comment by Jalle19** [comment]
@FernetMenta if the fix in https://github.com/Jalle19/xbmc/commit/3e02123e9d80ed1f409e360b5eebc89d3933c166 is correct, do you think it warrants backporting?

File: xbmc/cores/VideoRenderers/WinRenderer.cpp [hunk 1]
```diff
@@ -496,7 +496,7 @@ bool CWinRenderer::CreateIntermediateRenderTarget(unsigned int width, unsigned i
 
   if(!m_IntermediateTarget.Create(width, height, 1, D3D11_USAGE_DEFAULT, format))
   {
-    CLog::Log(LOGERROR, __FUNCTION__": intermediate render target creation failed.", format);
+    CLog::Log(LOGERROR, __FUNCTION__": intermediate render target creation failed");
```
Comment (afedchin) [inline xbmc/cores/VideoRenderers/WinRenderer.cpp]: I prefer message with point at the end. But this doesn't matter.

**Comment by afedchin** [comment]
+1 from me. thanks.

**Comment by mkortstiege** [comment]
Builds fine on Linux, but let's check the others as well. jenkins build this please

**Comment by Jalle19** [comment]
If it takes long for the pinged members to respond I can create a separate PR for https://github.com/Jalle19/xbmc/commit/ec094cc7df222d948d389fa38b5be8dfa358ecde, without it PVR is broken on Windows (possible on other platforms too).

File: xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp [hunk 2]
```diff
@@ -238,9 +238,10 @@ STDMETHODIMP_(ULONG) CDXVADecoderWrapper::AddRef(void)
 STDMETHODIMP_(ULONG) CDXVADecoderWrapper::Release(void)
 {
   AtomicDecrement(&m_refs);
-  assert(m_refs >= 0);
-  if (m_refs == 0) delete this;
-  return m_refs;
+  volatile long refs = m_refs;
```
Comment (Paxxi) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp]: This isn't thread safe, what if someone takes a reference after this line, we'll delete the object and go boom

Comment (Jalle19) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp]: Should I leave it unchanged instead?

Comment (Paxxi) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp]: Original isn't thread safe either but you can stay with original in this pr, it's a bit tricky to get right and I need to test it out

**Comment by MartijnKaijser** [comment]
nudge :)

**Comment by mkortstiege** [comment]
@Jalle19 please rebase ;)

**Comment by Jalle19** [comment]
@mkortstiege took care of the comments, dropped skeptical commits, rebased

**Comment by mkortstiege** [comment]
Thanks. @FernetMenta mind having a look?

**Comment by mkortstiege** [comment]
@FernetMenta, objections?

**Comment by FernetMenta** [comment]
could you drop a0c3f8045bb951b9affd287be4dff194f08291fc
I already eliminated the entire file

**Comment by Jalle19** [comment]
@FernetMenta dropped it

jenkins build this please

**Pull request merged**